gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -187,34 +235,97 @@ def handle_event(self, ev):
-                      (ev.id, service_type, str(err).capitalize()))
-            LOG.error(result)
-        finally:
-            msg = {'receiver': const.ORCHESTRATOR,
-                   'resource': resource,
-                   'method': ev.id,
-                   'kwargs': [
-                              {
-                               'context': context,
-                               'resource': resource,
-                               'request_info': request_info,
-                               'result': result
-                               }
-                            ]
-                   }
-            if not notification_data:
-                notification_data.update(msg)
-            else:
-                data = {
-                        'context': context,
-                        'resource': resource,
-                        'request_info': request_info,
-                        'result': result
-                        }
-                notification_data['kwargs'].append(data)
+            import traceback
+            traceback.print_exc()
+            LOG.error("Failed to process ev.id=%s, ev=%s reason=%s" % (ev.id,
+                                                                       ev.data,
+                                                                       err))
+            result = FAILED
+
+        if ev.id == const.EVENT_CONFIGURE_HEALTHMONITOR:
+            if (kwargs.get('periodicity') == INITIAL and
+                    result == SUCCESS):
+                notification_data = self._prepare_notification_data(ev, result)
+                self._sc.poll_event_done(ev)
+                self._notification(notification_data)
+            elif kwargs.get('periodicity') == FOREVER:
+                if result == FAILED:
+                    """If health monitoring fails continuously for 5 times
+                       send fail notification to orchestrator
+                    """
+                    kwargs['fail_count'] = kwargs.get('fail_count') + 1
+                    if kwargs.get('fail_count') >= MAX_FAIL_COUNT:
+                        notification_data = self._prepare_notification_data(
+                                                                        ev,
+                                                                        result)
+                        self._sc.poll_event_done(ev)
+                        self._notification(notification_data)
+                elif result == SUCCESS:
+                    """set fail_count to 0 if it had failed earlier even once
+                    """
+                    kwargs['fail_count'] = 0
+        elif ev.id == const.EVENT_CLEAR_HEALTHMONITOR:
+            """Stop current poll event. event.key is vmid which will stop
+               that particular service vm's health monitor
+            """
+            notification_data = self._prepare_notification_data(ev, result)
```

Comment:
Should we handle the case where the poll_event_done() is already invoked as a result of consecutive health check failures and again configurator gets clear_healthmonitor() API.  What will happen if poll_event_done() gets invoked for an unexisting event?